src/ctype/CTypeMetadata.spec.ts

```diff
@@ -0,0 +1,51 @@
+import CType from './CType'
+import ICType, { ICtypeMetadata } from '../types/CType'
+import CTypeMetadata from './CTypeMetadata'
+import * as CTypeUtils from './CTypeUtils'
+import { CTypeWrapperMetadata } from './CTypeSchema'
+
+jest.mock('../blockchainApiConnection/BlockchainApiConnection')
+
+describe('CType', () => {
+  const ctype = new CType({
+    schema: {
+      $id: 'http://example.com/ctype-1',
+      $schema: 'http://kilt-protocol.org/draft-01/ctype#',
+      properties: {
+        'first-property': { type: 'integer' },
+        'second-property': { type: 'string' },
+      },
+      type: 'object',
+    },
+  } as ICType)
+
+  const ctypeMetadata = {
+    metadata: {
+      $id: 'http://example.com/metadata-1',
+      $schema: 'http://kilt-protocol.org/draft-01/ctype#',
+      title: { type: 'string' },
+      description: { type: 'string' },
+      properties: {
+        'first-property': { type: 'integer' },
+        'second-property': { type: 'string' },
+      },
+      type: 'object',
+    },
+    ctypeHash: 'string',
+  } as ICtypeMetadata
+
```

Comment:
This is the general format I was going for with it, but I think it has room for improvement, maybe we don't need the $id or $schema. But, it could be a way to use it again. 

I could remove it altogether. Room for discussion

Reply:
Is this comment supposed to be at the Ctype creation?

Reply:
Done!